File: src/test/java/org/railwaystations/rsapi/utils/JwtUtilTest.java
package org.railwaystations.rsapi.utils;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.text.ParseException;

import static org.assertj.core.api.Assertions.assertThat;

class JwtUtilTest {

    @Test
    void generateWriteAndLoadRsaKey() throws IOException, ParseException {
        var temp = Files.createTempDirectory("rsapi");
        var rsaKeyFile = temp.resolve("rsaKeyFile.json");
        var rsaKey = JwtUtil.generateRsaKey();
        System.out.println(rsaKeyFile);
        System.out.println(rsaKey.toJSONString());
        JwtUtil.writeRsaKey(rsaKey, rsaKeyFile.toString());
        assertThat(rsaKeyFile).isNotEmptyFile();
        var loadedRsaKey = JwtUtil.loadRsaKey(rsaKeyFile.toString());
        assertThat(loadedRsaKey).isEqualTo(rsaKey);
    }

}

File: src/test/kotlin/org/railwaystations/rsapi/utils/JwtUtilTest.kt
package org.railwaystations.rsapi.utils

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.railwaystations.rsapi.utils.JwtUtil.generateRsaKey
import org.railwaystations.rsapi.utils.JwtUtil.loadRsaKey
import org.railwaystations.rsapi.utils.JwtUtil.writeRsaKey
import java.nio.file.Files

internal class JwtUtilTest {
    @Test
    fun generateWriteAndLoadRsaKey() {
        val temp = Files.createTempDirectory("rsapi")
        val rsaKeyFile = temp.resolve("rsaKeyFile.json")
        val rsaKey = generateRsaKey()
        println(rsaKeyFile)
        println(rsaKey.toJSONString())
        writeRsaKey(rsaKey, rsaKeyFile.toString())
        assertThat(rsaKeyFile).isNotEmptyFile()
        val loadedRsaKey = loadRsaKey(rsaKeyFile.toString())
        assertThat(loadedRsaKey).isEqualTo(rsaKey)
    }
}